docs/design/operator.md

```diff
@@ -0,0 +1,200 @@
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Cluster Upgrade
+
+The Gluster operator will be able to execute a rolling upgrade of any hosted or
+managed Gluster cluster. It will trigger an upgrade one node at a time, waiting
+for the node to come up and for the cluster to report no pending heals to any
+files before triggering a subsequent node.
+
+**Target Configurations:** Hosted, Managed
+
+### Monitoring
+
+To provide event-driven features, the Gluster operator will monitor various
+aspects of the clusters it will automate.
+
+#### Storage Capacity
+
+The Gluster operator will monitor the storage capacity of the overall cluster.
+Storage capacity is defined here as the amount of space that can be committed to
+new Gluster volumes and not how much free space is available. It will trigger
+events when storage capacity reaches "Warning" levels and "Critical" levels.
+These levels will be configurable, and should be set to a value that will allow
+for corrective action to take place before the storage reaches maximum capacity.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Volume Density
+
+The Gluster operator will monitor the total volume density of any cluster it is
+automating. It will trigger events when the number of volumes in the cluster
+reaches a pre-defined level that should prompt corrective action. This value
+will be configurable, and should be set to a value that will allow for
+corrective action to take place before cluster nodes lag or crash.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Device Health
+
+The Gluster operator will monitor the health of all devices in an automated
+Gluster cluster. It will report a "Warning" when a device reports failing health
+and will report a "Critical" problem when a device becomes inaccessible. Support
+for this feature may vary form one device type to another.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Volume Health
+
+The Gluster operator will monitor the health of all volumes in an automated
+Gluster cluster. It will report a "Warning" when a volume reports failing bricks
+and will report a "Critical" problem when a volume becomes inaccessible.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Node Health
+
+The Gluster operator will monitor the health of all nodes in an automated
+Gluster cluster. It will report a "Critical" problem when a node becomes
+inaccessible.
+
+**Target Configurations:** External, Hosted, Managed
+
+### Cluster Management
+
+In response to the monitoring events above, the Gluster operator will take any
+number of corrective management actions.
+
+#### Storage Scaling
+
+In the event that the overall cluster reports a critical capacity level, the
+operator will deploy a set number of additional storage nodes with new storage
+devices attached. The number of additional nodes to be deployed will be
+configurable.
+
+**Target Configurations:** Managed
+
+#### Cluster Scaling
```

Comment:
I think this should be out of scope at this time. GD2 should fix the need for multiple clusters, meaning we'd only need this if it falls short... a good reason to push it out.